manifests/charts/istio-control/istio-discovery/files/gateway-injection-template.yaml

```diff
@@ -12,10 +12,6 @@ metadata:
     {{ end }}
   }
 spec:
-  securityContext:
```

Comment:
IIRC, the PR that added this was merged recently. Deleting the security context will introduce a regression I think. Instead, we may be able to make this optional in case you have your own settings.

Reply:
If it were optional, I think it would solve _my_ specific use case since I'm already providing a securityContext but wouldn't benefit anyone else limited by a 3.x based kernel who isn't already specifying a securityContext, which IMO makes this setting a regression either way. Then to make Istio usable for those users, they either still have to upgrade their OS, or have something in the docs roughly equivalent to "just set _something_ so the gateway doesn't take privileges on your kernel settings". On the other hand by removing this, you could already provide a securityContext through existing installation methods, so users that needed to directly use port 80 or 443 for example could set this as needed, or alternatively map a port > 1024 in the service overrides

Reply:
cc related folks to take a look @hemendrateli @howardjohn

Reply:
I am ok with both ways.
Kernel version 4.11 was released around 7 years back (which added net.ipv4.ip_unprivileged_port_start feature). I am not sure how many users still are on kernel < 4.11

If we want to merge this PR then [these](https://github.com/istio/istio/issues/48364#issuecomment-1858453470) kind of users will be broken if they were using port < 1024.

so in any way I think we should update documentation also about broken users.

Reply:
> so users that needed to directly use port 80 or 443 for example could set this as needed, or alternatively map a port > 1024 in the service overrides.

I would very much prefer to make people binding to port 80/443 (99.99% of users) have a smoother experience than users running >7 year old kernels, if we have to make that tradeoff.

Reply:
From the original issue:
> I'm currently setting my securityContext anyway, which I would expect the chart to honor vs merging the two.

:point_up: that is also what I would expect to happen, and that is how I would prefer this to work, we cannot remove `securityContext` for everyone.

Reply:
IMO， These configurations should be the same as down：
https://github.com/istio/istio/blob/f653b34fe27ec8f781e9d837d43cd6234e3a34b6/manifests/charts/gateway/templates/deployment.yaml#L38-L45

Instead of forcing defaults everywhere, users can't use valuese support it at all

Reply:
Hi @nicole-lihui @bleggett, are there any pointers on what should be the `securityContext` for the users that still use kernel < 4.11?